src/controllers/fimsController.ts

```diff
@@ -0,0 +1,150 @@
+
+const responseErrorInternal = (reason: string) => (e: Error) =>
+  ResponseErrorInternal(`${reason} | ${e.message}`);
+
+/**
+ * Retrieves the access history for a given fiscal code.
+ *
+ * @param ioFimsService - The service used to fetch the access history.
+ * @param fiscalCode - The fiscal code for which the access history is requested.
+ * @param page - (Optional) The page number for paginated results.
+ * @returns A TaskEither containing either an error or the access history.
+ */
+const getAccessHistory = (
+  ioFimsService: IoFimsService,
+  fiscalCode: FiscalCode,
+  page?: string
+) =>
+  pipe(
+    TE.tryCatch(
+      () => ioFimsService.getAccessHistory(fiscalCode, page),
+      () => new Error("Error while fetching the access history")
+    ),
+    TE.mapLeft(responseErrorInternal("Fetching error"))
+  );
+
+/**
+ * Requests an export of data for a given fiscal code and email using the provided IoFimsService.
+ *
+ * @param ioFimsService - The service used to request the export.
+ * @param fiscalCode - The fiscal code of the user for whom the export is requested.
+ * @param email - The email address to which the export will be sent.
+ * @returns A TaskEither that resolves to an error or the result of the export request.
+ */
+const requestExport = (
+  ioFimsService: IoFimsService,
+  fiscalCode: FiscalCode,
+  email: EmailString
+) =>
+  pipe(
+    TE.tryCatch(
+      () => ioFimsService.requestExport(fiscalCode, email),
+      () => new Error("Error while requesting the export")
+    ),
+    TE.mapLeft(responseErrorInternal("Error while requesting the export"))
+  );
+
+export default class IoFimsController {
+  /**
+   * Constructs an instance of IoFimsController.
+   *
+   * @param ioFimsService - The service responsible for FIMS operations.
+   * @param profileService - The service responsible for profile operations.
+   */
+  constructor(
+    private readonly ioFimsService: IoFimsService,
+    private readonly profileService: ProfileService
+  ) {}
+
+  /**
+   * Retrieves the access history for a user.
+   *
+   * @param req - The express request object containing the user's request.
+   * @returns A promise that resolves to one of the following:
+   * - `IResponseErrorValidation` if there is a validation error.
+   * - `IResponseErrorInternal` if there is an internal server error.
+   * - `IResponseSuccessJson<AccessHistoryPage>` if the access history is successfully retrieved.
+   */
+  public readonly getAccessHistory = (
+    req: express.Request
+  ): Promise<
+    | IResponseErrorValidation
+    | IResponseErrorInternal
+    | IResponseSuccessJson<AccessHistoryPage>
+  > =>
+    withUserFromRequest(req, async (user) =>
+      pipe(
+        t.union([t.undefined, t.string]).decode(req.query.page),
+        TE.fromEither,
+        TE.altW(() => TE.right(undefined)),
+        TE.flatMap((page) =>
+          getAccessHistory(this.ioFimsService, user.fiscal_code, page)
+        ),
+        TE.toUnion
+      )()
+    );
```

Comment:
In the OpenAPI the validation error is specified as a `422` but if I am not mistaking this method will return a `400` because at the end we will call [this funciton](https://github.com/pagopa/io-ts-commons/blob/master/src/responses.ts#L330).

Reply:
https://github.com/pagopa/io-backend/commit/bd7f183cc58cdabff53de963559a5d0baa14f96f ✅